content/articles/stream-filter-in-java-using-lambda-expressions/index.md

```diff
@@ -0,0 +1,170 @@
+
+### Table of Contents
+- [Overview of the Stream Filter Method in Java](#overview-of-the-stream-filter-method-in-java)
+- [Using the Stream Filter Method](#using-the-stream-filter-method)
+- [Collection Filtering](#collection-filtering)
+- [Filtering Data Based on a Variety of Criteria](#filtering-data-based-on-a-variety-of-criteria)
+- [Methods for Dealing with Exceptions](#methods-for-dealing-with-exceptions)
+
+### Overview of the Stream Filter Method in java.
+The filter() function of the Java stream allows you to narrow down the stream's items based on a criterion. If you only want items that are even on your list, you can use the filter method to do this. This method accepts a predicate as an input and returns a list of elements that are the results of that predicate.
+
+It is possible to get a stream of items from this stream that meet a particular predicate with the filter() method. This is a step-between process. Performing an intermediate action like filter() does not really filter anything, but instead generates a new stream that, when browsed, includes the items of the first stream that satisfy the provided predicate. These operations are always lazy.
+
+### Using the Stream filter method
+Java stream offers the filter() method, which allows you to filter stream elements based on a predicate you specify. You can conveniently get only even elements from your list by using the filter method. This method accepts a predicate as a parameter and returns a stream of elements that are the results of the predicate.
+
+A Stream interface's filter() method identifies elements in a stream that satisfy a criterion. It is a Stream interface intermediate operation.
+
+The following is the method signature for the Stream filter() function:
+```Java
+Stream<q> filter(Predicate<? super q> predicate)
+```
+
+Notice how it accepts a Predicate object as a parameter. A predicate is a logical interface to a functional interface. Therefore, you may also send a lambda expression to this function.
+
+### Collection filtering
+The filter() function is often used to handle collections. We can use it to create a list of workers who have earned more than 90 marks, specifying the predicate as a lambda.
+
+```Java
+List<Employee> employeeWith90MarksAndAbove = employees
+  .stream()
+  .filter(q -> q.getMarks() > 90)
+  .collect(Collectors.toList());
+```
+Additionally, methods reference, which again is shorthand for such a lambda expression, may also be used.
+
+```Java
+List<Employee> employeeWith90MarksAndAbove = employees
+  .stream()
+  .filter(Employee::hasOverNinetyMarks)
+  .collect(Collectors.toList());
+```
+In this particular instance, we enhance our Employee class by adding the hasOverNinetyMarks method:
+
+```Java
+public boolean hasOverNinetyMarks() 
+{
+    return this.marks > 90;
+}
+```
+When we use these two approaches, we obtain the same result:
+
+```Java
+assertThis(employeeWith90MarksAndAbove).hasSize(3);

```

Comment:
Where does the `assertThis` method come from? There doesn't seem to be any method like that in the Java standard library. If it's from a maven dependency mention it in the guide to prevent any confusion. The reader shouldn't have to guess how the code should be run and if they need any dependencies for it.

Reply:
You should show the state of the original list within the code you used to filter it. How is the reader supposed to know that the result list should have a size of 3 and contain two objects `george` and `mike`? Also, show the declaration of the `george` and `mike` objects.